devex/vscode-extension/assets/edgeSolutionTemplateFolder/modules/[[project-name]]/host/main.c

```diff
@@ -0,0 +1,235 @@
+
+    char* enclaveMessage = (char*)malloc(512 * sizeof(char));
+    int enclaveResult = call_enclave(messageBodyStr, enclaveMessage, 512);
+    if (enclaveResult != 0)
+    {
+        result = IOTHUBMESSAGE_ABANDONED;
+    }
+    else
+    {
+        // This message should be sent to next stop in the pipeline, namely "output1".  What happens at "outpu1" is determined
+        // by the configuration of the Edge routing table setup.
+        MESSAGE_INSTANCE *messageInstance = CreateMessageInstance(enclaveMessage);
+        if (NULL == messageInstance)
+        {
+            result = IOTHUBMESSAGE_ABANDONED;
+        }
+        else
+        {
+            printf("Sending message (%zu) to the next stage in pipeline\n", messagesReceivedByInput1Queue);
+
+            clientResult = IoTHubModuleClient_LL_SendEventToOutputAsync(iotHubModuleClientHandle, messageInstance->messageHandle, "output1", SendConfirmationCallback, (void *)messageInstance);
+            if (clientResult != IOTHUB_CLIENT_OK)
+            {
+                IoTHubMessage_Destroy(messageInstance->messageHandle);
+                free(messageInstance);
+                printf("IoTHubModuleClient_LL_SendEventToOutputAsync failed on sending msg#=%zu, err=%d\n", messagesReceivedByInput1Queue, clientResult);
+                result = IOTHUBMESSAGE_ABANDONED;
+            }
+            else
+            {
+                result = IOTHUBMESSAGE_ACCEPTED;
+            }
+        }
+    }
+
+    free(enclaveMessage);
+    return result;
+}
+
+static IOTHUBMESSAGE_DISPOSITION_RESULT InputQueue1Callback(IOTHUB_MESSAGE_HANDLE message, void* userContextCallback)
+{
+    IOTHUB_MODULE_CLIENT_LL_HANDLE iotHubModuleClientHandle = (IOTHUB_MODULE_CLIENT_LL_HANDLE)userContextCallback;
+
+    unsigned char* messageBody;
+    size_t contentSize;
+
+    if (IoTHubMessage_GetByteArray(message, &messageBody, &contentSize) != IOTHUB_MESSAGE_OK)
+    {
+        messageBody = "<null>";
+    }
+
+    char* messageBodyStr = (char*)malloc(contentSize + 1);
+    memcpy(messageBodyStr, messageBody, contentSize);
+    messageBodyStr[contentSize] = '\0';
+
+    printf("Received Message [%zu]\r\n Data: [%s]\r\n",
+            messagesReceivedByInput1Queue, messageBodyStr);
+
+    IOTHUBMESSAGE_DISPOSITION_RESULT result = SendEnclaveResponse(iotHubModuleClientHandle, messageBodyStr);
+    messagesReceivedByInput1Queue++;
+    return result;
+}
+
+static IOTHUB_MODULE_CLIENT_LL_HANDLE InitializeConnection()
+{
+    IOTHUB_MODULE_CLIENT_LL_HANDLE iotHubModuleClientHandle;
+
+    if (IoTHub_Init() != 0)
+    {
+        printf("Failed to initialize the platform.\r\n");
+        iotHubModuleClientHandle = NULL;
+    }
+    else if ((iotHubModuleClientHandle = IoTHubModuleClient_LL_CreateFromEnvironment(MQTT_Protocol)) == NULL)
+    {
+        printf("ERROR: IoTHubModuleClient_LL_CreateFromEnvironment failed\r\n");
+    }
+    else
+    {
+        // Uncomment the following lines to enable verbose logging.
```

Comment:
Commented out code is not very maintainable... what if the signature changes?  You won't notice.
Replace with an ifdef, and make sure there's some test that verifies it builds with that define set.

Reply:
Now compiled in unconditionally, the developer can switch the global flag to enable tracing instead.